Sources/SwiftProtobuf/BinaryDecoder.swift

```diff
@@ -917,6 +917,12 @@ internal struct BinaryDecoder: Decoder {
                 try KeyType.decodeSingular(value: &k, from: &subdecoder)
             case 2: // Value is a message type
                 try subdecoder.decodeSingularEnumField(value: &v)
+                if v == nil {
```

Comment:
This was simpler than I thought it would be.

I don't suppose anything similar could ever apply to the key here?  I can't come up with any case that would.